src/pages/global-authorization/forms/AccessLevelForm.js

```diff
@@ -0,0 +1,114 @@
+// ** MUI Imports
+import { Grid } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
+import { useFormik } from 'formik'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { DataSets } from 'src/resources/DataSets'
+
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { AccessControlRepository } from 'src/repositories/AccessControlRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+
+export default function AccessLevelForm ({
+  labels,
+  maxAccess,
+  data,
+  moduleId
+})
+{
+
+    const [isLoading, setIsLoading] = useState(false)
+
+    const [initialValues, setInitialData] = useState({
+      accessLevel: '',
+      accessLevelName: ''
+   })
+
+    const { getRequest, postRequest } = useContext(RequestsContext)
+
+    const invalidate = useInvalidate({
+        endpointId: SystemRepository.ModuleClassRES.qry
+      })
+
+    const formik = useFormik({
+        initialValues,
+        enableReinitialize: true,
+        validateOnChange: true,
+        validationSchema: yup.object({
+          //accessLevel: yup.string().required('')
+
+        }),
+        onSubmit: async obj => {
+
+          const updatedData = data.list.map(item => ({
+                moduleId:moduleId,
+                resourceId: item.key,
+                accessLevel: obj.accessLevel
+            })
+          )
+
+          updatedData.forEach(item => {
+            item.accessLevel != "" ? item.accessLevel = item.accessLevel : item.accessLevel = 0
```

Comment:
Confusing. Should be `item.accessLevel = item.accessLevel || 0` in case we want to revert undefined, null, and empty string to 0.